Objective-C/CBLValueIndexConfiguration.h

```diff
@@ -28,10 +28,22 @@ NS_ASSUME_NONNULL_BEGIN
 @interface CBLValueIndexConfiguration : CBLIndexConfiguration
 
 /**
- Constructor for creating a value index by using an array of expression strings.
+ A predicate expression defining conditions for indexing documents.
+ Only documents satisfying the predicate are included, enabling partial indexes.
+ */
+@property (nonatomic, readonly, nullable) NSString* where;
+
+/**
+ Initializes a value index by using an array of expression strings.
  */
 - (instancetype) initWithExpression: (NSArray<NSString*>*)expressions;
 
+/**
+ Initializes a partial value index with an array of expression strings and a where clause.
+ */
+- (instancetype) initWithExpression: (NSArray<NSString*>*)expressions
+                              where: (nullable NSString*)where;
+
```

Comment:
Sorry, I have suggested that comment. After reading it, it doesn't feel right as the where is optional. Probably something like this?

```
Initializes a value index with an array of expression strings and an optional where clause for a partial index.
```

Reply:
Changed this to explicit optional. I have also now added API description for the constructors and its params